### PR TITLE
Ci 3197 os UI autocomplete component index stories tsx add client side

### PR DESCRIPTION
**Pull request by mocca102**
Adding tests to cover these scenarios

When passed
-  API key
- ConstructorIO Client 
- CIOClientOptions
- Custom placeholder
- Custom styles
- Fully featured example

Screenshot 2024-06-14 at 6 43 04 PM

### CODE REVIEW

**Review by VincentGalloro — APPROVED** [review]
LGTM

File: spec/Components/CioAutocomplete/CioAutocomplete.server.test.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it.only("Throws an error even if we provide a ConstructorIO Client since it's not SSR supported", () => {
```
Comment (esezen) [inline spec/Components/CioAutocomplete/CioAutocomplete.server.test.jsx:20]: Only

**Review by esezen — APPROVED** [review]
One test seems to be failing but I believe we have a story for that, right? The test itself seems fine, the implementation needs to be updated

**Comment by mocca102** [comment]
> One test seems to be failing but I believe we have a story for that, right? The test itself seems fine, the implementation needs to be updated

The tests needed to be updated on this PR. Please review the updated tests @esezen

**Review by esezen — APPROVED** [review]
LGTM! You might want to rebase or merge the other branch into this since you made some changes to the other one

**Pull request merged**